ui/app/hooks/useMixpanelEvent.js

```diff
@@ -0,0 +1,8 @@
+import { useContext, useCallback } from 'react'
+import { MixPanelContext } from '../contexts/metametrics'
+
+export function useMixPanelEvent (eventName, data) {
+  const mixPanelEvent = useContext(MixPanelContext)
+  const trackEvent = useCallback(() => mixPanelEvent(eventName, data), [eventName, data, mixPanelEvent])
```

Comment:
So this `useCallback` call is intended to ensure the function instance remains the same between renders, right? Would it make sense to use this in `MixPanelProvider` instead, so that the `mixPanelEvent` function instance remains unchanged between renders for all uses of the provider, not just for this hook?